### PR TITLE
Terraform best practices

### PR DESCRIPTION
**Pull request by htahir1**
## Describe changes
Preview: https://zenml-io.gitbook.io/hamza-test/how-to/infrastructure-deployment/infrastructure-as-code/terraform-stack-management  and https://zenml-io.gitbook.io/hamza-test/how-to/infrastructure-deployment/infrastructure-as-code/best-practices

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docs/book/how-to/infrastructure-deployment/infrastructure-as-code/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+```
+
+These practices help maintain a clean, scalable, and maintainable infrastructure codebase while following infrastructure-as-code best practices. Remember to:
+
+- Keep configurations DRY using locals and variables
+- Use consistent naming conventions across resources
+- Document all required configuration fields
+- Consider component dependencies when organizing stacks
+- Separate infrastructure and ZenML registration state
+- Use workspaces for different environments
```
Comment (strickvl) [inline docs/book/how-to/infrastructure-deployment/infrastructure-as-code/best-practices.md:464]: link to workspace docs, or did you mean TF workspaces?

File: docs/book/how-to/infrastructure-deployment/infrastructure-as-code/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,469 @@
...
+- Keep registration state with the ML operations team
```
Comment (strickvl) [inline docs/book/how-to/infrastructure-deployment/infrastructure-as-code/best-practices.md:465]: what does this mean?

File: docs/book/how-to/infrastructure-deployment/infrastructure-as-code/terraform-stack-management.md [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+}
+```
+
+To generate a API key, use the command:
+
+```bash
+zenml service-account create <SERVICE_ACCOUNT_NAME>
+```
+
+You can learn more about how to generate a ZENML_API_KEY via service accounts
```
Comment (strickvl) [inline docs/book/how-to/infrastructure-deployment/infrastructure-as-code/terraform-stack-management.md:73]:
```suggestion
You can learn more about how to generate a 1ZENML_API_KEY` via service accounts
```

**Review by strickvl — CHANGES_REQUESTED** [review]
Looks pretty good! just some small nits

File: docs/book/how-to/infrastructure-deployment/infrastructure-as-code/terraform-stack-management.md [hunk 4]
```diff
@@ -70,7 +70,7 @@ To generate a API key, use the command:
 zenml service-account create <SERVICE_ACCOUNT_NAME>
 ```
 
-You can learn more about how to generate a ZENML_API_KEY via service accounts
+You can learn more about how to generate a 1ZENML_API_KEY` via service accounts
```
Comment (strickvl) [inline docs/book/how-to/infrastructure-deployment/infrastructure-as-code/terraform-stack-management.md:73]:
```suggestion
You can learn more about how to generate a `ZENML_API_KEY` via service accounts
```

**Review by strickvl — APPROVED** [review]

**Review by stefannica — APPROVED** [review]

**Pull request merged**